### ISSUE TITLE
lego ignores retry-after timer included in order polling response

### ISSUE THREAD

**Issue by grindsa** [go-acme/lego#1601]
### Welcome

- [X] Yes, I've searched similar issues on GitHub and didn't find any.

### How do you use lego?

Binary

### Detailed Description

Hi,

I am using lego to test a [non-Letsencrypt server implementation](https://github.com/grindsa/acme2certifier)

Platform is Ubuntu 18.04, I am using your [lego 4.6.0 binary](https://github.com/go-acme/lego/releases/download/v4.6.0/lego_v4.6.0_linux_amd64.tar.gz)

Certificate enrolment except for situations in which certificate issuance gets delayed by CA server.
In such case the my acme-proxy will set the status of the order resource to “processing”, returns a “Retry-After” header as part of the response to client and expects client to send a new polling request after this period (as described RFC 8555 section 7.4).

Based on my logs I see that lego ignores the "Retry-After" timer and polls the order resource 0.5second interval for 30seconds. If the CA server will not issue a certificate within this period the certificate enrolment will fail.

I know that the timeout can be adjusted by setting the `--cert.timeout` flag but do you see any chance set the polling interval based on the retry-after value? 

Thank you for your help and have a nice day.
/GrindSa

**Comment by azlux** [comment]
I think I need that too. Some CA are slow, and lego stop polling before the CA have create the certificate.

**Comment by ldez** [comment]
@azlux I don't think this is related: I guess you have a timeout problem, this is different than the `Retry-After` header.
Please open a dedicated discussion: https://github.com/go-acme/lego/discussions/categories/q-a

**Comment by azlux** [comment]
Hi @ldez , I was speaking about the `Retry-after` header as describe here: https://www.rfc-editor.org/rfc/rfc8555.html#section-8.2
When my CA take 5min to generate the certificate, I don't need lego to smash my acme server during 30s every second since the server know it will not have the certificate in time.
